receiver/fluentforwardreceiver/conversion.go

```diff
@@ -160,25 +162,53 @@ func parseRecordToLogRecord(dc *msgp.Reader, lr pdata.LogRecord) error {
 			return msgp.WrapError(err, "Record", key)
 		}
 
-		// fluentd uses message, fluentbit log.
-		if key == "message" || key == "log" {
-			switch v := val.(type) {
-			case string:
-				lr.Body().SetStringVal(v)
-			case []uint8:
-				// Sometimes strings come in as uint8's.
-				lr.Body().SetStringVal(string(v))
-			default:
-				return fmt.Errorf("cannot convert message type %T to string", val)
-			}
+		s, err := ValueToString(val)
+		if err != nil {
+			return err
+		}
+
+		if Contains(conf.Mappings.Body, key) {
+			insertToAttributeMap(key, val, &bodyMap)
```

Comment:
By simplicity, I try to do the best to convert a maximum of fields in the body, se I work on body attributemap instead of string value.